### PR TITLE
Allow running services without a gateway

### PR DESCRIPTION
**Pull request by jvstme**
Behind the PROXY feature flag.
Part of #1595

### CODE REVIEW

File: src/dstack/_internal/server/services/gateways/__init__.py
Comment (jvstme) [inline src/dstack/_internal/server/services/gateways/__init__.py:1]: The `services/gateways` directory needs rearranging - with the new terms, only some of this logic is gateway-specific, while some is proxy-specific. Will do in another PR.

**Review by r4victor — APPROVED** [review]
